src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -112,15 +138,16 @@ export const Configurator = ({
                     text: field.displayName || field.name,
                     value: field.name,
                   })),
-                  { text: '-----', value: '', disabled: true },
-                  { text: 'No time field', value: undefined },
+                  { text: '-----', value: '-----', disabled: true },
+                  { text: noTimeFilter, value: noTimeFilter },
                 ]}
                 value={timeFieldName}
                 onChange={(e) => {
-                  const value = e.target.value === 'undefined' ? undefined : e.target.value;
-                  setTimeFieldName(value);
+                  const value = e.target.value === noTimeFilter ? undefined : e.target.value;
+                  setTimeFieldName(e.target.value);
```

Comment:
how come not `value`?

Reply:
`e.target.value` refers to the `text` displayed, so it won't actually be set as the `value` prop for the option.

Reply:
this seem like a bug